main/src/ca/uwaterloo/flix/language/errors/WeederError.scala

```diff
@@ -389,6 +389,29 @@ object WeederError {
     def explain(formatter: Formatter): Option[String] = None
   }
 
+  /**
+    * An error raised to indicate an unsupported restrictable choose rule.
+    *
+    * @param summary  a summary of the error.
+    * @param short    a short name for the arrow pointing to the location.
+    * @param loc      the location where the error occurs.
+    */
+  case class UnsupportedRestrictableChooseRule(summary: String, short: String, loc: SourceLocation) extends WeederError {
```

Comment:
It's unusual that we have the summary etc. so customizable but maybe that's better for experimentation.